### PR TITLE
Fix potential Null pointer dereference in test-time.c

### PR DESCRIPTION
**Pull request by icy17**

### CODE REVIEW

File: test/test-time.c [hunk 1]
```diff
@@ -104,6 +104,9 @@ main(int argc, char **argv)
 
 	for (i = 0; i < NEVENT; i++) {
 		ev[i] = evtimer_new(base, time_cb, event_self_cbarg());
+		if (ev[i] == NULL) {
```
Comment (azat) [inline test/test-time.c:107]: You need to add cleanup path, otherwise events allocated on the previous iterations will leak

Comment (icy17) [reply test/test-time.c:107]: OK, I think I should also free the event_base.

File: test/test-time.c [hunk 2]
```diff
@@ -117,10 +117,19 @@ main(int argc, char **argv)
...
 		i, called, NEVENT);
 
+	
 	if (i == 1 && called >= NEVENT) {
 		return EXIT_SUCCESS;
 	} else {
 		return EXIT_FAILURE;
 	}
+err:
+	if (base)
```
Comment (azat) [inline test/test-time.c:127]: events should be freed first, otherwise this will cause use-after-free

**Comment by azat** [comment]
Merged in https://github.com/libevent/libevent/pull/1681. Thanks!

**Pull request closed without merging**